Adding performance and data science concepts (#8893)
conda/conda: docs/source/user-guide/concepts/conda-performance.rst

index metadata is always growing, and thus conda becomes slower as the
number of packages increases.

How a package is installed
==========================

While you are waiting, conda is doing a lot of work installing the 
packages. At any point along these steps, performance issues may arise.

Conda follows these steps when installing a package:

#. Downloading and processing index metadata
#. Reducing the index
#. Expressing the package data and constraints as a SAT problem
#. Running the solver
#. Downloading and extracting packages
#. Verifying package contents
#. Linking packages from package cache into environments

Therefore, if you're experiencing a slowdown, evaluate the following questions
to identify potential causes:

* Are you creating a new environment or installing into an existing one?
* Does your environment have pip-installed dependencies in it?
* What channels are you using?
* What packages are you installing?
* Is the channel metadata sane?
* Are channels interacting in bad ways?

Improving conda performance
===========================

To address these challenges, you can move packages to archive

> channels and follow the methods below to present a reduced view of

Reviewer: ```suggestion
channels and follow the methods below to present conda with a smaller, simpler view than
```

        * Create a fresh environment. As environments grow, they become harder
          and harder to solve. Working with small, dedicated environments can
          be much faster.

Set strict channel priority
---------------------------

Setting strict channel priority makes it so that if a package exists on
a channel, conda will ignore all packages with the same name on lower
priority channels.

.. figure:: ../../img/strict-disabled.png
    :width: 50%
.. figure:: ../../img/strict-enabled.png
    :width: 50%

This can dramatically reduce package search space and reduces the use of
improperly constrained packages.

One thing to consider is that setting strict channel priority may make
environments unsatisfiable. Learn more about :ref:`strict`.

Reduce the index
----------------
One option for speeding up conda is to reduce the index. It's likely that
your repodata contains package data that is not used in the solving stage.
Filtering out unnecessary packages before solving can save time.
Reducing the index takes specs and reduces the problem size so that the
SAT doesn’t have as much work to do.

> Additionally, it is beneficial to be as specific as possible in your package

Reviewer: Reducing the index is something that conda does based on the user's input specs.  Making your inputs more specific makes this index reduction more effective, and thus makes the process faster.  Users alone do not reduce the index, but they should understand why it's important to think about specs.
